File: mns-core/src/main/java/cz/fb/manaus/core/category/categorizer/CountryCodeCategorizer.java
package cz.fb.manaus.core.category.categorizer;

import cz.fb.manaus.core.model.Market;
import org.springframework.stereotype.Component;

import java.util.Optional;
import java.util.Set;

@Component
final public class CountryCodeCategorizer extends AbstractDelegatingCategorizer {

    public static final String PREFIX = "country_";

    public CountryCodeCategorizer() {
        super(PREFIX);
    }

    @Override
    protected Set<String> getCategoryRaw(Market market) {
        var countryCode = market.getEvent().getCountryCode();
        return Set.of(Optional.ofNullable(countryCode)
                .map(String::toLowerCase)
                .orElse("none"));
    }
}

File: mns-core/src/main/java/cz/fb/manaus/core/category/categorizer/CountryCodeCategorizer.kt
package cz.fb.manaus.core.category.categorizer

import cz.fb.manaus.core.model.Market
import org.springframework.stereotype.Component

const val COUNTRY_PREFIX = "country_"

@Component
object CountryCodeCategorizer : AbstractDelegatingCategorizer(COUNTRY_PREFIX) {

    override fun getCategoryRaw(market: Market): Set<String> {
        return when (val countryCode = market.event.countryCode) {
            null -> emptySet()
            else -> setOf(countryCode.toLowerCase())
        }
    }
}
